catalog/forms.py

```diff
@@ -0,0 +1,23 @@
+import datetime
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data['renewal_date']
```

Comment:
attribute `cleaned_data` lưu trữ data dạng gì? data được xử lý ntn?

Reply:
attribute `cleaned_data` lưu trữ data ở dạng dictionary. Dữ liệu sẽ được làm sạch và loại bỏ input có khả năng không an toàn bằng cách sử dụng validator mặc định và chúng cũng được chuyển đổi về dạng chuẩn của dữ liệu.